scipy/cluster/tests/test_vq.py

```diff
@@ -289,3 +289,11 @@ def test_kmeans_large_thres(self):
         res = kmeans(x, 1, thresh=1e16)
         assert_allclose(res[0], np.array([4.]))
         assert_allclose(res[1], 2.3999999999999999)
+        
+    def test_kmeans2_kpp_low_dim(self):
+        # Regression test for gh-11462
+        prev_res = np.array([[-1.95266667,  0.898],
+                             [-3.153375  ,  3.3945]])
+        np.random.seed(42)
+        res, _ = kmeans2(TESTDATA_2D, 2, minit='++')
```

Comment:
This probably makes sense since you are testing that a non-problematic case is consistent with before.

That said, a true regression guard for the issue would also look at a data set that will trigger the problem on current master.

Since `(200, 2)` is the shape of `TESTDATA_2D`, you'll also want a test case
where `N > M`, and your original example with shape `(5, 1000)` might be suitable for that.

Reply:
Ok, I appreciate the comments.

I added the test case where ```N > M```.